backend/dataall/modules/shares_base/services/sharing_service.py

```diff
@@ -356,6 +362,48 @@ def reapply_share(cls, engine: Engine, share_uri: str) -> bool:
                     if not lock_released:
                         log.error(f'Failed to release lock for dataset {share_data.dataset.datasetUri}.')
 
+    @staticmethod
+    def fetch_pending_shares(engine):
+        """
+        A method used by the scheduled ECS Task to run fetch_pending_shares() process against ALL shared objects in ALL
+        active share objects within dataall
+        """
+        with engine.scoped_session() as session:
+            pending_shares = (
+                session.query(ShareObject)
+                .join(
+                    Notification,
+                    and_(
+                        ShareObject.shareUri == func.split_part(Notification.target_uri, '|', 1),
+                        ShareObject.datasetUri == func.split_part(Notification.target_uri, '|', 2),
+                    ),
+                )
+                .filter(and_(Notification.type == 'SHARE_OBJECT_SUBMITTED', ShareObject.status == 'Submitted'))
+                .all()
+            )
+            return pending_shares
+
+    @staticmethod
+    def _get_share_data(session, uri):
+        share = ShareObjectRepository.get_share_by_uri(session, uri)
+        dataset = DatasetRepository.get_dataset_by_uri(session, share.datasetUri)
+        share_items_states = ShareObjectRepository.get_share_items_states(session, uri)
```

Comment:
ShareObjectRepository.get_share_items_states does not exist anymore

Reply:
Done